### PR TITLE
* added base isSameMatch() and identicalTo() Object Matcher

### PR DESCRIPTION
**Pull request by oneronaut**
* added an isSameMatch() function to all Matchers, to allow checking for identical matching criteria (useful for Rokito integration).

* Added an identicalTo() Object Matcher for checking objects are identical in structure and value without necessarily being the same instance.

### CODE REVIEW

File: tests/source/brsHamcrest/Test_brsHamcrest_Helpers.brs [hunk 1]
```diff
@@ -190,3 +190,152 @@ sub test_IsEnumerable_false (t as Object)
...
+            knownBoolean: true
+        }
+    end function
+    'WHEN'
+    result = coreDoMatch(getFooObj(), getDifferentFooObj())
+    'THEN'
+    t.assertFalse(result)
+
+    teardown_brsHamcrest_Helpers()
+end sub
```
Comment (imbenjamin) [inline tests/source/brsHamcrest/Test_brsHamcrest_Helpers.brs:244]: Can we have a test that compares when two objects have the same keys but different values? Eg:
```
objectA = { value1: "a", value2: "b", value3: "c" }
objectB = { value1: "x", value2: "y", value3: "z" }
```

File: tests/source/brsHamcrest/brsHamcrest_Matchers/Test_brsHamcrest_ObjectMatchers.brs [hunk 2]
```diff
@@ -111,3 +111,55 @@ sub test_sameObjectAs_targetAndValueAreNotObjects (t as Object)
...
+    result = identicalTo(testValue).doMatch(testTarget)
+
+    'THEN'
+    t.assertFalse(result)
+
+    teardown_brsHamcrest_ObjectMatchers()
+end sub
+
+
+sub test_identicalTo_targetAndValueAreIdenticalObjects (t as Object)
```
Comment (imbenjamin) [inline tests/source/brsHamcrest/brsHamcrest_Matchers/Test_brsHamcrest_ObjectMatchers.brs:134]: This test is failing on my device (Roku 3 - 4200X, FW 7.6.0)
```
------ Running dev 'Unit Tests' main ------
###################################################
                Running unit tests!
###################################################
..........................................................................................F...........................
======================================================================
pkg:/source/brsHamcrest/brsHamcrest_Matchers/Test_brsHamcrest_ObjectMatchers.brs
FAIL: test_identicalTo_targetAndValueAreIdenticalObjects
----------------------------------------------------------------------
expression evaluates to false

----------------------------------------------------------------------
Ran 118 tests

FAILED (
failures= 1
)

###################################################
                Test suite complete
###################################################

```

**Review by imbenjamin — CHANGES_REQUESTED** [review]
One test is failing, plus a test is missing I think. Also, RokuUtils.pyc, TestrunnerUtils.pyc and __init__.pyc should be excluded from the commit as they're compiled Python files.

File: tests/source/brsHamcrest/brsHamcrest_Matchers/Test_brsHamcrest_ObjectMatchers.brs (hunk 2 above)
Comment (oneronaut) [reply tests/source/brsHamcrest/brsHamcrest_Matchers/Test_brsHamcrest_ObjectMatchers.brs:134]: Weird. Passing on my 7.5.3 box. I'll look into it.

Comment (oneronaut) [reply tests/source/brsHamcrest/brsHamcrest_Matchers/Test_brsHamcrest_ObjectMatchers.brs:134]: Actually, it wasn't passing. I missed it.

File: tests/source/brsHamcrest/Test_brsHamcrest_Helpers.brs (hunk 1 above)
Comment (oneronaut) [reply tests/source/brsHamcrest/Test_brsHamcrest_Helpers.brs:244]: Added one new test to cover this.

File: tests/source/brsHamcrest/brsHamcrest_Matchers/Test_brsHamcrest_ObjectMatchers.brs (hunk 2 above)
Comment (oneronaut) [reply tests/source/brsHamcrest/brsHamcrest_Matchers/Test_brsHamcrest_ObjectMatchers.brs:134]: Now passing and pyc files removed.

**Review by imbenjamin — APPROVED** [review]

**Comment by imbenjamin** [comment]
Thanks, merging...

**Pull request merged**